### PR TITLE
Implement cyclic burst time generation

### PR DESCRIPTION
**Pull request by loostrum**

### CODE REVIEW

File: frbpoppy/time_dists.py [hunk 1]
```diff
@@ -31,12 +31,54 @@ def regular(rate=2, n_srcs=1, n_days=1, z=0):
...
+    frac = np.atleast_1d(frac)
+    z = np.atleast_1d(z)
+
+    # get number of bursts to generate for each source
+    nburst_per_cycle = (rate * frac * period).astype(int)
+    nburst_per_source = ((n_days / period) * nburst_per_cycle).astype(int)
+
+    # generate burst arrival times within each active period
+    times = np.random.uniform(0, frac[:, np.newaxis] * period[:, np.newaxis],
+                              (n_srcs, nburst_per_source.max())).astype(np.float32)
```
Comment (davidgardenier) [inline frbpoppy/time_dists.py:63]: There's still a bug in here, as this generates unsorted bursts.

Comment (davidgardenier) [reply frbpoppy/time_dists.py:63]: Any chance you still would have time to look into this? :)

Comment (loostrum) [reply frbpoppy/time_dists.py:63]: I'll have a look; for most of the function it doesn't matter that the bursts aren't sorted, but it does for the redshift calculation and probably the output should be sorted anyway?

**Comment by davidgardenier** [comment]
Not sure it necessarily is important in advance of the redshift
calculation, but the output array will need to be sorted. Perhaps that is
almost the easiest implementation?

**Comment by davidgardenier** [comment]
Actually, in advance of the redshift is important for determining t0...

**Pull request merged**